pkg/services/hcloud/server/server.go

```diff
@@ -158,7 +211,7 @@ func (s *Service) handleBootStateUnset(ctx context.Context) (reconcile.Result, e
 	}
 
 	// if provider id is not set create the server.
```

Comment:
can you remove this comment? It doesn't really fit anymore, because you use this state in all cases... The providerID will pretty much never be set